app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -370,6 +372,7 @@ class BrowserRobot {
         }
 
         fun openHomeScreen(interact: HomeScreenRobot.() -> Unit): HomeScreenRobot.Transition {
+            sessionLoadedIdlingResource = SessionLoadedIdlingResource()
```

Comment:
In addition to this, you missed this part: 
runWithIdleRes(sessionLoadedIdlingResource) {
                onView(
                        anyOf(
                                ViewMatchers.withResourceName("mozac_browser_toolbar_browser_actions")
                        )
                )
                        .check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
            }

You should have to add that too...otherwise I guess it does not make any difference

Reply:
Okay